### PR TITLE
[Merged by Bors] - feat: infinite normed field is noncompact

### PR DESCRIPTION
**Pull request by urkud**
Generalize from nontrivially normed fields to normed fields.
Also prove that a nontrivially normed field is infinite.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/Basic.lean [hunk 1]
```diff
@@ -341,22 +342,52 @@ protected theorem NormedSpace.unbounded_univ : ¬Bornology.IsBounded (univ : Set
...
+  rw [neBot_iff, Ne.def, cobounded_eq_bot_iff, ← isBounded_univ]
+  exact NormedSpace.unbounded_univ 𝕜 E
+
+instance (priority := 100) NontriviallyNormedField.cobounded_neBot : NeBot (cobounded 𝕜) :=
+  NormedSpace.cobounded_neBot 𝕜 𝕜
+
+instance (priority := 100) RealNormedSpace.cobounded_neBot [NormedSpace ℝ E] :
+    NeBot (cobounded E) := NormedSpace.cobounded_neBot ℝ E
+
+instance (priority := 100) NontriviallyNormedField.infinite : Infinite 𝕜 :=
```
Comment (sgouezel) [inline Mathlib/Analysis/NormedSpace/Basic.lean:355]: Could you lower a little bit the priority of these, to 90 or 80? It looks like there are several more reasonable options to be tried before.

File: Mathlib/Analysis/NormedSpace/Basic.lean [hunk 2]
```diff
@@ -341,22 +342,52 @@ protected theorem NormedSpace.unbounded_univ : ¬Bornology.IsBounded (univ : Set
...
+This cannot be an instance because in order to apply it,
+Lean would have to search for `NormedSpace 𝕜 E` with unknown `𝕜`.
+We register this as an instance in two cases: `𝕜 = E` and `𝕜 = ℝ`. -/
+protected theorem NormedSpace.noncompactSpace : NoncompactSpace E := by
+  by_cases H : ∃ c : 𝕜, c ≠ 0 ∧ ‖c‖ ≠ 1
+  · letI := NontriviallyNormedField.ofNormNeOne H
+    exact ⟨fun h ↦ NormedSpace.unbounded_univ 𝕜 E h.isBounded⟩
+  · push_neg at H
+    rcases exists_ne (0 : E) with ⟨x, hx⟩
+    suffices ClosedEmbedding (Infinite.natEmbedding 𝕜 · • x) from this.noncompactSpace
```
Comment (AntoineChambert-Loir) [inline Mathlib/Analysis/NormedSpace/Basic.lean:375]: That would be good to have these closed embeddings in general? and even in finite dimension.

File: Mathlib/Analysis/NormedSpace/Basic.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Analysis/NormedSpace/Basic.lean]: Done.

File: Mathlib/Analysis/NormedSpace/Basic.lean (hunk 2 above)
Comment (urkud) [reply Mathlib/Analysis/NormedSpace/Basic.lean:375]: What general statement are you talking about? https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/Module/FiniteDimension.html#closedEmbedding_smul_left ? Another statement is that a trivially normed field has discrete topology (so, in fact we can skip `Infinite.natEmbedding 𝕜`).

Comment (AntoineChambert-Loir) [reply Mathlib/Analysis/NormedSpace/Basic.lean:375]: I was thinking of the `suffices` that you prove on the next lines. I guess it also holds when the field is nontrivially valued. 
Actually, I am not sure you need to make the cases disjoint: once you reduce to the case of a line by these closed embeddings, what matters is that `k` is noncompact.

Comment (urkud) [reply Mathlib/Analysis/NormedSpace/Basic.lean:375]: If `k = Q`, then this scalar multiplication is not a **closed** embedding.

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Review by YaelDillies — COMMENTED** [review]
Looks good!

maintainer merge

**Pull request closed without merging**